src/app/scripts/cac/control/cac-control-directions.js

```diff
@@ -421,6 +427,31 @@ CAC.Control.Directions = (function (_, $, moment, Control, Routing, UserPreferen
         urlRouter.updateUrl(urlRouter.buildDirectionsUrlFromPrefs());
     }
 
+    /** Helper to save current directions origin or destination to user preferences.
+     *
+     * @param location {String} Either 'origin' or 'destination'
+     */
+    function updateLocationPreference(location) {
```

Comment:
I feel like the behavior of this is a bit not initially obvious, and maybe some naming changes would help.  Seeing `updateLocationPreference(location)`, my first thought would be that this is going to save a location stored in `location` to user preferences.  I think renaming the argument to something that makes it more obvious that it's the key rather than the actual location would help (I can't think of anything better than `key` off hand, though that's not ideal).  And though it would make the function name a little ridiculous, calling it `updateLocationPreferenceFromDirections` would make it more obvious that the source of the location is the `directions` variable on the parent scope.

Reply:
Since user preference has a helper method called `setLocation` which is how we generally set `origin` and `destination`, this seemed like a reasonable name. Added the docstring to clarify the param usage, but sure, can rename the param too.